### PR TITLE
Test: Unit test for Reconcile work controller

### PR DESCRIPTION
**Pull request by aagusuab**
### Description of your changes

Fixes #

I have:

- [x] Read and followed Caravel's [Code of conduct](https://github.com/Azure/k8s-work-api/blob/master/code-of-conduct.md).
- [x] Run `make reviewable` to ensure this PR is ready for review.

### CODE REVIEW

File: pkg/controllers/apply_controller_unit_test.go [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+			GroupVersionKind: testDeployment.GroupVersionKind(),
+			Scope:            nil,
+		}, nil
+	} else {
+		return nil, errors.New("test error: mapping does not exist.")
+	}
+}
+
+func TestReconcile(t *testing.T) {
+	workNamespace := getRandomString()
```
Comment (Ealianis) [inline pkg/controllers/apply_controller_unit_test.go:84]: No need to reinvent these string utils. Use the k8s API machiner utilities.
"k8s.io/apimachinery/pkg/util/rand"

```workNamespace := utilRand.String(5)```

File: pkg/controllers/apply_controller_unit_test.go [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+	workv1alpha1 "sigs.k8s.io/work-api/pkg/apis/v1alpha1"
+)
+
+var (
+	fakeDynamicClient = fake.NewSimpleDynamicClient(runtime.NewScheme())
+	appliedWork       = &workv1alpha1.AppliedWork{}
+	ownerRef          = metav1.OwnerReference{
+		APIVersion: workv1alpha1.GroupVersion.String(),
+		Kind:       appliedWork.Kind,
+		Name:       appliedWork.GetName(),
```
Comment (Ealianis) [inline pkg/controllers/apply_controller_unit_test.go:37]: Does the name / UID even need to set? They are being set to empty fields. The applymanifest func in the controller will set the ownerref, I think we need a test or an assert within a test to ensure the ownerref matches the expected AppliedWork.

https://github.com/Azure/k8s-work-api/blob/master/pkg/controllers/apply_controller.go#L160-L161

File: pkg/controllers/apply_controller_unit_test.go (hunk 1 above)
Comment (Ealianis) [inline pkg/controllers/apply_controller_unit_test.go:84]: These test setups are quite lengthy and make the function explode. IMO this makes it harder to read. Consider using a wrapper struct and helper functions to define the different states... 

A small example:
https://github.com/Ealianis/k8s-work-api/blob/710e7df6acbc07f6fc0c38f46559f299d43be183/pkg/controllers/finalize_controller_unit_test.go#L26-L27

File: pkg/controllers/apply_controller_unit_test.go (hunk 2 above)
Comment (aagusuab) [reply pkg/controllers/apply_controller_unit_test.go]: I did not need the Name and the UID as you said.
However, I have kept these, in order to test the ownerReference being equal shown here:
https://github.com/Azure/k8s-work-api/pull/26/files line 293.

File: pkg/controllers/apply_controller_unit_test.go (hunk 1 above)
Comment (aagusuab) [reply pkg/controllers/apply_controller_unit_test.go]: I have replaced all the custom functions with rand.String(5). (same library)

**Review by Ealianis — APPROVED** [review]
LGTM

**Pull request merged**